### DISCUSSION TITLE
Tap to add type

### DISCUSSION THREAD

**Discussion by maxlestage** [rust-lang/rust-analyzer#15784] [Q&A] [answered]
Good evening,

I allow myself to send you a message because I am surprised to have seen a useful feature disappear from the Rust Analyzer extension, indeed for a while I could visualize the inferred type of a variable for example and click on it to add it hard in my code. However, since some updates it is no longer possible to find this feature which as a beginner on the language was very useful to identify the type of things and have indications. And then it made it possible to add the type on the fly without having to write it. Is this change voluntary, how can I reactivate this tap parameter to add the type to a variable? Good to you and have a good evening, thanking you in advance for your help.

**Comment by alibektas** [comment]


**Reply by maxlestage** [reply]
Sorry for the somewhat late response, it seems that since version: 0.4.1653 of the Rust Analyzer extension, the "Double-click to insert" option has been removed from the extension's behavior. the following versions no longer include this feature and the vscode editor obviously does not provide an equivalent for rust in particular. (Not using TypeScript I don't know at all if it works for this language natively) finally I find it a shame that this feature is removed because it was useful.

**Comment by lnicola** [comment]
Sounds like a duplicate of https://github.com/rust-lang/rust-analyzer/issues/15604, which should have been fixed. What version are you actually running?

Just tested this, it works for me.

**Reply by maxlestage** [reply]
I'm struggling to understand how to update it, I can't change the extension code?

**Reply by lnicola** [reply]
Which version are you running and how did you install it?

**Reply by maxlestage** [reply]
I just installed the latest extension of vscode Rust Analyzer, and I have stable release of Rust v1.73, I haven't installed rust-analyzer from Brew command lines (I use a Mac)

**Reply by maxlestage** [reply]
```json
//This is my config settings.json in vscode :

{
    "files.autoSave": "onFocusChange",
    "editor.fontSize": 14,
    "editor.fontFamily": "Victor Mono",
    "editor.formatOnPaste": true,
    "editor.formatOnSave": true,
    "editor.formatOnType": true,
    "editor.fontLigatures": false,
    "editor.inlayHints.enabled": "on",
    "editor.guides.bracketPairs": true,
    "editor.inlineSuggest.enabled": true,
    "terminal.integrated.enableMultiLinePasteWarning": false,
    "workbench.colorTheme": "Material Theme Darker High Contrast",
    "editor.minimap.enabled": false,
    "rust-analyzer.completion.fullFunctionSignatures.enable": true,
    "rust-analyzer.completion.limit": null,
    "rust-analyzer.hover.documentation.enable": true,
    "rust-analyzer.hover.actions.implementations.enable": true,
    "rust-analyzer.hover.actions.enable": true,
    "rust-analyzer.hover.actions.gotoTypeDef.enable": true,
    "rust-analyzer.hover.actions.run.enable": true,
    "rust-analyzer.inlayHints.typeHints.enable": true,
    "rust-analyzer.inlayHints.parameterHints.enable": true,
    "rust-analyzer.hover.actions.references.enable": true,
    "rust-analyzer.inlayHints.bindingModeHints.enable": true,
    "rust-analyzer.inlayHints.closureCaptureHints.enable": true,
    "rust-analyzer.inlayHints.closureReturnTypeHints.enable": "with_block",
    "rust-analyzer.inlayHints.discriminantHints.enable": "never",
    "rust-analyzer.inlayHints.expressionAdjustmentHints.enable": "never",
    "rust-analyzer.inlayHints.lifetimeElisionHints.enable": "never",
    "rust-analyzer.inlayHints.lifetimeElisionHints.useParameterNames": false,
    "rust-analyzer.highlightRelated.references.enable": true,
    "rust-analyzer.inlayHints.maxLength": null,
    "rust-analyzer.inlayHints.typeHints.hideClosureInitialization": false,
    "rust-analyzer.lens.enable": true,
    "rust-analyzer.lens.references.adt.enable": true,
    "rust-analyzer.lens.references.enumVariant.enable": true,
    "rust-analyzer.lens.references.method.enable": true,
    "rust-analyzer.lens.references.trait.enable": true,
    "workbench.iconTheme": "material-icon-theme",
    "git.autofetch": true,
    "git.confirmSync": false,
    "telemetry.telemetryLevel": "off",
    "rust-analyzer.diagnostics.previewRustcOutput": true,
    "rust-analyzer.diagnostics.useRustcErrorCode": true,
    "rust-analyzer.inlayHints.chainingHints.enable": false,
    "editor.inlineSuggest.showToolbar": "always",
    "editor.inlineSuggest.suppressSuggestions": true,
}
```

**Reply by lnicola** [reply]
So are you actually on v0.3.1697? Do the inlay hints show up? Are you seeing this tooltip?

![image](https://github.com/rust-lang/rust-analyzer/assets/308347/16b4d8cf-23a3-4de7-b00d-040ede0e5b49)

**Reply by maxlestage** [reply]
<img width="2056" alt="Screenshot 2023-10-21 at 16 46 47" src="https://github.com/rust-lang/rust-analyzer/assets/56438697/9df3b320-7116-42a7-85f8-f607c432a85b">

**Reply by maxlestage** [reply]
It was possible to click on the type until the version: 0.4.1653, the following no longer integrate this feature.

**Reply by lnicola** [reply]
Why are you on v0.4.1653? Try picking v0.4.1702 from that list, or choose "Switch to release version" from the extension details page.

**Reply by maxlestage** [reply]
Because the latest version does not allow the click on the type, I just went to: v0.4.1702. 
I was able to install a less recent version from this menu:
<img width="602" alt="Screenshot 2023-10-21 at 17 18 31" src="https://github.com/rust-lang/rust-analyzer/assets/56438697/4c06565c-83e5-4072-b3ba-e9af47b92619">

**Reply by maxlestage** [reply]
I tried several versions before coming across the most recent one that still manages the desired feature

**Reply by lnicola** [reply]
Did you actually test that it's not working after upgrading to v0.4.1702?

Are you seeing the inlay hints, and if yes, what tooltip do you see when hovering over a type hint?

**Reply by maxlestage** [reply]
OK, let's look together I installed the last update I deleted my type so that the extension deduces it:
<img width="823" alt="Screenshot 2023-10-21 at 17 23 39" src="https://github.com/rust-lang/rust-analyzer/assets/56438697/7494b93c-704c-41db-8424-2c522925538a">

As you can see, I don't have the possibility of double click to add

**Reply by lnicola** [reply]
Which versions of RA and Code are you running and what happens if you double-click on the hint?

**Reply by maxlestage** [reply]
Version: 1.84.0-insider
Commit: 8e11b537e3b091a8b52c7179fb623ecdeecaf4ea
Date: 2023-10-20T05:36:15.501Z
Electron: 25.9.1
ElectronBuildId: 24472542
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Darwin arm64 22.6.0


<img width="1021" alt="Screenshot 2023-10-21 at 17 30 55" src="https://github.com/rust-lang/rust-analyzer/assets/56438697/3f437d25-6a17-4fae-96f7-60ccbdded791">


Currently I'm using the latest version of both

**Reply by maxlestage** [reply]
When I double click, nothing happens.

**Reply by maxlestage** [reply]
So I have no idea how to add this feature on the latest version

**Comment by lnicola** [comment]
For now, you can use the stable version of VS Code and type insertion will work.

**Reply by lnicola** [reply, marked as answer]
#15786 will fix this and will be available in the next nightly.

That said, if you're just learning the language, it's pretty unusual to specify every type like in your earlier screenshot. Especially in Code where you can see the exact types most of the time.

**Reply by maxlestage** [reply]
Thank you very much for the update, I just updated the extension and it integrates the functionality well.

I use VScode Insiders for Rust, and the VScode stable version for everything else at school. It allows me to have two different configurations. Thank you!
